File: 백준/Silver/1931. 회의실 배정/회의실 배정.js
const input = require("fs").readFileSync(process.platform === "linux" ? "/dev/stdin" : "알고리즘/example.txt").toString().trim().split("\n").map((el) => el.split(" ").map(Number));
const N = input[0][0];
const meetings = [...input.slice(1)];
// 빨리 끝나는 애들 기준으로 정렬 하고 끝나는 시간이 같다면 시작 시간 빠른 애가 앞 쪽에 정렬된다
meetings.sort((a, b) => a[1] - b[1] || a[0] - b[0])

function calc_meeting(arr){
    let count = 0;
    let time = 0;
    for(let i = 0; i < meetings.length; i++){
        // 현재 시간보다 회의 시작 시간이 늦어야지만 회의실 배정이 가능하다.
        if(time <= meetings[i][0]){
            // 회의 끝난 시간이 이제 현재 시간이 된다.
            time = meetings[i][1];
            count++;
        }
    }
    console.log(count);
}

calc_meeting(meetings);

File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
import sys
input = sys.stdin.readline

N = int(input())

meeting_list = []
for i in range(N):
    meeting_list.append(list(map(int, input().split())))

# 빨리 끝나는 애들 먼저 빨리 빨리 시켜줘야 많이 회의할 수 있다.
meeting_list.sort(key=lambda x: (x[1], x[0]))

count = 0
time = 0

for i in range(len(meeting_list)):
    if time <= meeting_list[i][0]:  
        count += 1
        time = meeting_list[i][1] 
        
print(count)